src/main/scala/org/broadinstitute/dsde/workbench/leonardo/db/DbReference.scala

```diff
@@ -0,0 +1,84 @@
+package org.broadinstitute.dsde.workbench.leonardo.db
+
+import java.sql.SQLTimeoutException
+
+import akka.actor.ActorSystem
+import com.google.common.base.Throwables
+import com.typesafe.config.Config
+import com.typesafe.scalalogging.LazyLogging
+import liquibase.{Contexts, Liquibase}
+import liquibase.database.jvm.JdbcConnection
+import liquibase.resource.{ClassLoaderResourceAccessor, ResourceAccessor}
+import org.broadinstitute.dsde.workbench.leonardo.config.LiquibaseConfig
+import slick.basic.DatabaseConfig
+import slick.dbio.DBIO
+import slick.jdbc.{JdbcBackend, JdbcDataSource, JdbcProfile, TransactionIsolation}
+import net.ceedubs.ficus.Ficus._
+import sun.security.provider.certpath.SunCertPathBuilderException
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object DbReference extends LazyLogging {
+
+  private def initWithLiquibase(dataSource: JdbcDataSource, liquibaseConfig: LiquibaseConfig, changelogParameters: Map[String, AnyRef] = Map.empty): Unit = {
+    val dbConnection = dataSource.createConnection()
+    try {
+      val liquibaseConnection = new JdbcConnection(dbConnection)
+      val resourceAccessor: ResourceAccessor = new ClassLoaderResourceAccessor()
+      val liquibase = new Liquibase(liquibaseConfig.changelog, resourceAccessor, liquibaseConnection)
+
+      changelogParameters.foreach { case (key, value) => liquibase.setChangeLogParameter(key, value) }
+      liquibase.update(new Contexts())
+    } catch {
+      case e: SQLTimeoutException =>
+        val isCertProblem = Throwables.getRootCause(e).isInstanceOf[SunCertPathBuilderException]
+        if (isCertProblem) {
+          val k = "javax.net.ssl.keyStore"
+          if (System.getProperty(k) == null) {
+            logger.warn("************")
+            logger.warn(s"The system property '${k}' is null. This is likely the cause of the database connection failure.")
+            logger.warn("************")
+          }
+        }
+        throw e
+    } finally {
+      dbConnection.close()
+    }
+  }
+
+  def init(config: Config, system: ActorSystem)(implicit executionContext: ExecutionContext): DbReference = {
+    val db = "mysql"
+    val dbConfig = DatabaseConfig.forConfig[JdbcProfile](db, config)
+
+    system.registerOnTermination {
+      dbConfig.db.shutdown
+    }
+
+    val liquibaseConf = config.as[LiquibaseConfig]("liquibase")
+    if (liquibaseConf.initWithLiquibase)
+      initWithLiquibase(dbConfig.db.source, liquibaseConf)
+
+    DbReference(dbConfig)
+  }
+}
+
+case class DbReference(private val dbConfig: DatabaseConfig[JdbcProfile])(implicit val executionContext: ExecutionContext) extends AllComponents {
```

Comment:
Oh shit, you're right. I should have read the Rawls comment. Slick does give you a thread pool, the Rawls one is on top of that. You can probably disregard this whole comment chain... ;)

A slick thread:
```
"slick.db-176" #269 daemon prio=5 os_prio=0 tid=0x00007fe5b4013000 nid=0x123 runnable [0x00007fe4b1699000]
   java.lang.Thread.State: RUNNABLE
        at java.net.SocketInputStream.socketRead0(Native Method)
        at java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
        at java.net.SocketInputStream.read(SocketInputStream.java:171)
        at java.net.SocketInputStream.read(SocketInputStream.java:141)
        at sun.security.ssl.InputRecord.readFully(InputRecord.java:465)
        at sun.security.ssl.InputRecord.read(InputRecord.java:503)
        at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:973)
        - locked <0x0000000090acfd40> (a java.lang.Object)
        at sun.security.ssl.SSLSocketImpl.readDataRecord(SSLSocketImpl.java:930)
        at sun.security.ssl.AppInputStream.read(AppInputStream.java:105)
        - locked <0x0000000090ad7d10> (a sun.security.ssl.AppInputStream)
        at com.mysql.jdbc.MysqlIO.readFully(MysqlIO.java:2954)
        at com.mysql.jdbc.MysqlIO.reuseAndReadPacket(MysqlIO.java:3375)
        at com.mysql.jdbc.MysqlIO.reuseAndReadPacket(MysqlIO.java:3365)
        at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3805)
        at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2478)
        at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2625)
        at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2547)
        - locked <0x0000000090ab4e70> (a com.mysql.jdbc.JDBC4Connection)
        at com.mysql.jdbc.ConnectionImpl.rollbackNoChecks(ConnectionImpl.java:4730)
        at com.mysql.jdbc.ConnectionImpl.rollback(ConnectionImpl.java:4622)
        - locked <0x0000000090ab4e70> (a com.mysql.jdbc.JDBC4Connection)
        at com.zaxxer.hikari.proxy.ConnectionProxy.rollback(ConnectionProxy.java:318)
        at com.zaxxer.hikari.proxy.ConnectionJavassistProxy.rollback(ConnectionJavassistProxy.java)
        at slick.driver.JdbcActionComponent$Rollback$$anonfun$run$2.apply$mcV$sp(JdbcActionComponent.scala:54)
        at slick.jdbc.JdbcBackend$BaseSession.endInTransaction(JdbcBackend.scala:443)
        at slick.driver.JdbcActionComponent$Rollback$.run(JdbcActionComponent.scala:54)
        at slick.driver.JdbcActionComponent$Rollback$.run(JdbcActionComponent.scala:52)
        at slick.backend.DatabaseComponent$DatabaseDef$$anon$2.liftedTree1$1(DatabaseComponent.scala:237)
        at slick.backend.DatabaseComponent$DatabaseDef$$anon$2.run(DatabaseComponent.scala:237)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
```

A Rawls DataSource thread:
```
"pool-1-thread-51" #453 prio=5 os_prio=0 tid=0x00007fe5580b6000 nid=0x1d3 waiting on condition [0x00007fe55c2fa000]
   java.lang.Thread.State: WAITING (parking)
        at sun.misc.Unsafe.park(Native Method)
        - parking to wait for  <0x00000000dd370008> (a scala.concurrent.impl.Promise$CompletionLatch)
        at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:836)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:997)
        at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1304)
        at scala.concurrent.impl.Promise$DefaultPromise.tryAwait(Promise.scala:202)
        at scala.concurrent.impl.Promise$DefaultPromise.ready(Promise.scala:218)
        at scala.concurrent.impl.Promise$DefaultPromise.result(Promise.scala:223)
        at scala.concurrent.Await$$anonfun$result$1.apply(package.scala:190)
        at scala.concurrent.BlockContext$DefaultBlockContext$.blockOn(BlockContext.scala:53)
        at scala.concurrent.Await$.result(package.scala:190)
        at org.broadinstitute.dsde.rawls.dataaccess.SlickDataSource$$anonfun$inTransaction$1.apply(DataSource.scala:65)
        at scala.concurrent.impl.Future$PromiseCompletingRunnable.liftedTree1$1(Future.scala:24)
        at scala.concurrent.impl.Future$PromiseCompletingRunnable.run(Future.scala:24)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
```